homeassistant/setup.py

```diff
@@ -685,10 +689,31 @@ def async_pause_setup(
     try:
         yield
     finally:
-        time_taken = time.monotonic() - started
         integration, group = running
-        # Add negative time for the time we waited
-        _setup_times(hass)[integration][group][phase] = -time_taken
+        if running not in _setup_started(hass):
+            # Suppose a task is started from inside the `async_start_setup` context
+            # manager. In that case, we will see the `current_setup_group`, but the
+            # context manager may have exited by that time if it's running in the
+            # background and not being awaited. So we also check if the integration
+            # group is still in the `setup_started`, as that indicates whether the
+            # context manager is still running. If it's not in `setup_started`, the
+            # context manager has exited, and we should not subtract the time.
+            _LOGGER.debug(
+                "%s (%s) finished while waiting for %s", integration, group, phase
+            )
+            # There is not return here since we are in a finally and do not
+            # want to swallow exceptions
+        else:
+            time_taken = time.monotonic() - started
+            # Add negative time for the time we waited
+            _setup_times(hass)[integration][group][phase] = -time_taken
+            _LOGGER.debug(
+                "Adding wait for %s for %s (%s) of %.2f",
+                phase,
+                integration,
+                group,
+                -time_taken,
```

Comment:
```suggestion
                time_taken,
```
Debug logging is for humans